tests/components/mqtt/common.py

```diff
@@ -66,6 +66,254 @@
+    },
+}
+MOCK_SUBENTRY_NOTIFY_COMPONENT_NO_NAME = {
+    "notify_none": {
+        "platform": "notify",
+        "qos": 0,
+        "command_topic": "test-topic",
+        "command_template": "{{ value_json.value }}",
+        "entity_picture": "https://example.com/notify_none",
+        "retain": False,
+    },
+}
+
+MOCK_SUBENTRY_SENSOR_COMPONENT = {
+    "sensor_bla789": {
+        "platform": "sensor",
+        "name": "Test sensor",
+        "qos": 1,
+        "state_topic": "test-topic3",
+        "entity_picture": "https://example.com/sensor_bla789",
+    },
+}
+
+MOCK_SUBENTRY_LIGHT_COMPONENT = {
+    "light_bla0ab": {
+        "platform": "light",
+        "name": "Test light",
+        "qos": 1,
+        "command_topic": "test-topic4",
+        "schema": "basic",
+        "entity_picture": "https://example.com/light_bla0ab",
+    },
+}
+
+MOCK_SUBENTRY_LIGHT_BAD_SCHEMA = {
+    "light_bla0ab": {
+        "platform": "light",
+        "name": "Test light",
+        "qos": 1,
+        "command_topic": "bad#topic",
+        "schema": "basic",
+    },
+}
+
+MOCK_NOTIFY_SUBENTRY_DATA_MULTI = {
+    "device": {
+        "name": "Milk notifier",
+        "sw_version": "1.0",
+        "hw_version": "2.1 rev a",
+        "model": "Model XL",
+        "model_id": "mn002",
+        "configuration_url": "https://example.com",
+    },
+    "components": MOCK_SUBENTRY_NOTIFY_COMPONENT1 | MOCK_SUBENTRY_NOTIFY_COMPONENT2,
+}
+
+MOCK_NOTIFY_SUBENTRY_DATA_SINGLE = {
+    "device": {
+        "name": "Milk notifier",
+        "sw_version": "1.0",
+        "hw_version": "2.1 rev a",
+        "model": "Model XL",
+        "model_id": "mn002",
+        "configuration_url": "https://example.com",
+    },
+    "components": MOCK_SUBENTRY_NOTIFY_COMPONENT1,
+}
+MOCK_SUBENTRY_DATA_NOTIFY_NO_NAME = {
+    "device": {
+        "name": "Milk notifier",
+        "sw_version": "1.0",
+        "hw_version": "2.1 rev a",
+        "model": "Model XL",
+        "model_id": "mn002",
+        "configuration_url": "https://example.com",
+    },
+    "components": MOCK_SUBENTRY_NOTIFY_COMPONENT_NO_NAME,
+}
+
+MOCK_SUBENTRY_DATA_SET_MIX = {
+    "device": {
+        "name": "Milk notifier",
+        "sw_version": "1.0",
+        "hw_version": "2.1 rev a",
+        "model": "Model XL",
+        "model_id": "mn002",
+        "configuration_url": "https://example.com",
+    },
+    "components": MOCK_SUBENTRY_NOTIFY_COMPONENT1
+    | MOCK_SUBENTRY_NOTIFY_COMPONENT2
+    | MOCK_SUBENTRY_SENSOR_COMPONENT
+    | MOCK_SUBENTRY_LIGHT_COMPONENT,
+}
+
+MOCK_SUBENTRY_DATA_BAD_COMPONENT_SCHEMA = {
+    "device": {
+        "name": "Milk notifier",
+        "sw_version": "1.0",
+        "hw_version": "2.1 rev a",
+        "model": "Model XL",
+        "model_id": "mn002",
+        "configuration_url": "https://example.com",
+    },
+    "components": MOCK_SUBENTRY_LIGHT_BAD_SCHEMA,
+}
+
+MOCK_SUBENTRY_NOTIFY_COMPONENT1 = {
+    "notify_milkman_alert": {
```

Comment:
I think the key is now a UUID, right? Maybe better use some test-UUID here for more realistic tests?